client/product_details/AddToBag.jsx

```diff
@@ -39,30 +44,69 @@ class AddToBag extends React.Component {
     });
   }
   
-  handleQuantitySubmit(e) {
+  handleQuantityChange(e) {
     var selectedQuantity = e.target.value;
     this.setState({
       selectedQuantity: selectedQuantity
     });
   }
 
+  handleAddToBagSubmit() {
+    if (this.state.selectedSize !== 'SELECT SIZE') {
+      for (var i = 0; i < this.props.availableSizes.length; i++) {
+        if (this.props.availableSizes[i][0] === this.state.selectedSize) {
+          var sku_id = this.props.availableSizes[i][2];
+          console.log('styleId: ', sku_id);
+        }
+      }
+      var options = {
+        "sku_id": sku_id,
+        "count": this.state.selectedQuantity
+      };
+      // TODO: Add call to post data here & to server
+      fetch(`http://localhost:8080/cart`, {
+        method: 'post',
+        headers: {'Content-Type': 'application/json'},
+        body: JSON.stringify(options)
+      })
+        .then((response) => {
+          console.log(response.status);
+        })
+        .catch((error) => {
+          console.log(error);
+        });
+    } else {
+      // TODO: Add code to open the size dropdown
+      this.setState({
+        showMessage: true
+      });
+    }
+  }
+
   render() {
     var quantityArr = [];
     for (var i = 1; i < (this.state.maxQuantity + 1); i++) {
       quantityArr.push(i);
     }
-    
+    var defaultSizeOption = this.props.availableSizes.length === 0 ? 'OUT OF STOCK' : 'SELECT SIZE';
+    var defaultQuantity = this.state.selectedSize === 'SELECT SIZE' ? '-' : 1;
+
     return (
       <div id="addToBag">
-        <select name={this.state.selectedSize} id="selectSize" placeholder="SELECT SIZE" onChange={this.handleSizeSubmit}>
-          <option value="SELECT SIZE">SELECT SIZE</option>
+        {this.state.showMessage ? <a id="errorMessage">Please Select Size</a> : null}
+        <br></br>
+        <select name={this.state.selectedSize} id="selectSize" placeholder={defaultSizeOption} onChange={this.handleSizeChange}>
+          <option value={defaultSizeOption}>{defaultSizeOption}</option>
           {this.props.availableSizes.map((size) => <option value={size[0]}>{size[0]}</option>)}
         </select>
-        <select name={this.state.selectedQuantity} id="selectQuantity" placeholder={this.state.selectedQuantity} onChange={this.handleQuantitySubmit}>
+        <select name={this.state.selectedQuantity} id="selectQuantity" placeholder={defaultQuantity} onChange={this.handleQuantityChange}>
+          {this.state.selectedSize === 'SELECT SIZE' ? <option value="-">-</option> : null}
```

Comment:
I've never implemented the logic this way in React before! Very neat! Seems very handy :)

Reply:
Yes! Very useful :)